features/filesystem/fat/FATFileSystem.cpp

```diff
@@ -160,6 +160,24 @@ void ff_memfree(void *p)
 }
 
 // Implementation of diskio functions (see ChaN/diskio.h)
+static WORD disk_get_sector_size(BYTE pdrv)
+{
+    WORD ssize = _ffs[pdrv]->get_erase_size();
+    if (ssize < 512) {
+        ssize = 512;
+    }
+
+    MBED_ASSERT(ssize >= _MIN_SS && ssize <= _MAX_SS);
```

Comment:
I think the assertions should be moved in the constructor, something like:
```c
MBED_ASSERT(bd->get_program_size() == bd->get_read_size());
```
The erase size should then be checked by `is_valid_erase` so the optimization kicks in when it's possible to do so and should return the _real_ size instead of returning `_block_size`.